### DISCUSSION TITLE
Does SHM make no sense?

### DISCUSSION THREAD

**Discussion by wddpfx** [eProsima/Fast-DDS#4120] [General]
There are three kind of transport:TCP、UDP and SHM.

we can use SHM in the same host. But there is DataSharing module.

we can transport data using DataSharing in the same host, just in form of CHANGE，without serializing or deserializing data by Message module.
so what is the meaning of SHM?

sorry my English is not good. Can I be understood?

**Comment by JesusPoderoso** [comment]
Hi @wddpfx , thanks for using Fast DDS.
I would recommend you to read the [DataSharing section](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/datasharing.html) of the Fast DDS documentation, [its constraints](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/datasharing.html#constraints), and the difference in the amount of copies performed in both cases (avoiding encapsulation) described in [this discussion thread](https://github.com/eProsima/Fast-DDS/discussions/3654).

Although Data-sharing delivery uses shared memory, it differs from Shared Memory Transport in that Shared Memory is a full-compliant transport. That means that with Shared Memory Transport the data being transmitted must be copied from the DataWriter history to the transport and from the transport to the DataReader. With Data-sharing these copies can be avoided.

[Shared Memory Transport](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/shared_memory/shared_memory.html) has the same behavior as UDP or TCP transport (data is serialized and encapsulated, with the amount of copies that this entails), but instead of writting/reading the final data from a socket, it writes/reads it from a memory segment.

**Reply by wddpfx** [reply]
OK think you.
